staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -2038,6 +2070,37 @@ func TestAllocator(t *testing.T) {
 				deviceAllocationResult(req0SubReq1, driverA, pool1, device1, false),
 			)},
 		},
+		"device-binding-conditions": {
+			deviceBinding: true,
+			claimsToAllocate: objects(
+				claimWithRequests(claim0, nil, request(req0, classA, 1))),
+			classes: objects(class(classA, driverA)),
+			slices:  objects(sliceWithOneDeviceAndBindingConditions(slice1, node1, pool1, driverA, []string{"IsPrepare"}, []string{"BindingFailed"})),
+			node:    node(node1, region1),
+
+			expectResults: []any{
+				resourceapi.AllocationResult{
+					Devices: resourceapi.DeviceAllocationResult{
+						Results: []resourceapi.DeviceRequestAllocationResult{
+							deviceRequestAllocationResult(req0, driverA, pool1, device1, ptr.To(false), []string{"IsPrepare"}, []string{"BindingFailed"}, nil),
+						},
+					},
+					NodeSelector: localNodeSelector(node1),
+				},
+			},
+		},
+		"node-restriction": {
+			deviceBinding:    true,
+			claimsToAllocate: objects(claim(claim0, req0, classA)),
+			classes:          objects(class(classA, driverA)),
+			slices:           objects(sliceWithOneDeviceAndRestrictedUsage(slice1, localNodeSelector(node1), pool1, driverA)),
```

Comment:
Shouldn't we have a test where we start out with a node selector that indicates a broad set of nodes, but because the node restriction is there, what ends up in the result is just the chosen node? It's not clear to me that is what this is testing?